internal/migrations/20241122160000_update_new_column_control_plane_count_value_for_existing_cluster_records.go

```diff
@@ -0,0 +1,52 @@
+package migrations
+
+import (
+	gormigrate "github.com/go-gormigrate/gormigrate/v2"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+// split to 2 migrations, one for each query.
+
+func updateNewColumnControlPlaneCountValueForExistingClusterRecords() *gormigrate.Migration {
+	migrate := func(tx *gorm.DB) error {
+		transaction := tx.Begin()
+
+		cleanQuery := tx.Session(&gorm.Session{NewDB: true})
+		err := transaction.Model(&common.Cluster{}).
+			// control_plane_count value in existing records can be NULL or 0 (default). We want to set the value in both cases
+			Where(cleanQuery.Where("control_plane_count IS NULL").Or("control_plane_count = ?", 0)).
+			Where("high_availability_mode = ?", models.ClusterCreateParamsHighAvailabilityModeNone).
+			Update("control_plane_count", "1").Error
+		if err != nil {
+			transaction.Rollback()
+			return errors.Wrap(err, "failed to update control_plane_count value of existing SNO clusters to 1")
+		}
+
+		cleanQuery = tx.Session(&gorm.Session{NewDB: true})
+		err = transaction.Model(&common.Cluster{}).
+			// control_plane_count value in existing records can be NULL or 0 (default). We want to set the value in both cases
+			Where(cleanQuery.Where("control_plane_count IS NULL").Or("control_plane_count = ?", 0)).
+			Where("high_availability_mode = ?", models.ClusterCreateParamsHighAvailabilityModeFull).
+			Update("control_plane_count", "3").Error
```

Comment:
What will happen if we have a cluster without hosts and the high availability is full? Do we still want to update it to 3, or should it be 0?

Reply:
We want it to be 3 as it represents the desired amount of hosts

Reply:
Why we need need to migrate installed clusters?